app/api/adapters/db/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""
+Database adapter for Postgres database.
+This module is an abstraction around the psycopg3 python postgres library. It
+has methods for connecting to the database, and abstracts away the loading of
+config and environment variables.
+
+Usage:
+    # During application initialization
+    from api.adapters import db
+    db.init()
+
+    # In a route handler
+    from api.adapters import db
+    with db.get_connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute("SELECT 1")
+"""
+
+import atexit
+import os
+
+import psycopg
+import psycopg_pool
+from psycopg import rows
+
+from api import logging
+
+# Re-export these types to restrict the API of this module
+# to only the types we want to expose explicitly.
+Connection = psycopg.Connection
+Cursor = psycopg.Cursor
+DBConnectionError = psycopg.OperationalError
+kwargs_row = rows.kwargs_row
+
+logger = logging.get_logger(__name__)
+
+
+def init():
+    """Initializes the database connection pool"""
+    global pool
+    print(get_conninfo())
+    pool = psycopg_pool.ConnectionPool(get_conninfo(), open=False)
+
+    # Open the connection pool when the app starts
+    pool.open()
+
+    # # Close the connection pool when the app shuts down
+    atexit.register(pool.close)
+
+
+def get_conninfo() -> str:
+    """Gets a db connection info based on environment variables"""
+    dbhost = os.getenv("DB_HOST")
+    dbname = os.getenv("POSTGRES_DB")
+    username = os.getenv("POSTGRES_USER")
+    password = os.getenv("POSTGRES_PASSWORD")
+    ENVIRONMENT = os.getenv("ENVIRONMENT", "production")
```

Comment:
Rather than os.getenv, could make a model derived from `PydanticBaseEnvConfig` which loads it for you.